### PR TITLE
Add save inference model to trainer

### PR DESCRIPTION
**Pull request by jacquesqiao**

### CODE REVIEW

**Comment by luotao1** [comment]
可以加一个单测，用high level api训练的模型 （示例在 `python/paddle/fluid/tests/book/high-level-api/`）,用底层api来预测（示例在 `python/paddle/fluid/tests/book/`）。

**Review by panyx0718 — COMMENTED** [review]
test?

**Comment by jacquesqiao** [comment]
@panyx0718 https://github.com/PaddlePaddle/Paddle/pull/12682/files#diff-b8b5b08b124f579a337e2b89165ad65e

File: python/paddle/fluid/trainer.py [hunk 1]
```diff
@@ -431,6 +431,28 @@ def save_params(self, param_path):
...
+                             target_var_indexes):
+        """
+        Save model for cpp inference into :code:`param_path`.
+
+        Args:
+            param_path(str): The path to save parameters.
+            feeded_var_names(list(str)): The name of the vars that you
+                need to feed in before run program.
+            target_var_indexes(list(int)): the index of target var that
+                you need to return in trainer.train_func.
```
Comment (panyx0718) [inline python/paddle/fluid/trainer.py:444]: not sure if user know how to use it...I'm hoping that Trainer/Inferencer are in contrib namespace

**Review by panyx0718 — APPROVED** [review]

**Review by luotao1 — APPROVED** [review]
LGTM

File: python/paddle/fluid/trainer.py (hunk 1 above)
Comment (jacquesqiao) [reply python/paddle/fluid/trainer.py:444]: I agree. Currently, Trainer is lack of maintenance.

**Review by typhoonzero — APPROVED** [review]
Approve API

**Pull request merged**